### ISSUE TITLE
Problem when trying the tf classifier on different dataset

### ISSUE THREAD

**Issue by rafaa1988** [kalaspuffar/tensorflow-data#3]
I have to say that your example is one of the best I found in Tensorflow, especially the tf data. I'm using your classifier on my dataset. Attached 2 pictures for 2 different classes (1,2). First class has 1000 pictures. Second class has 400 pictures. The difference between the two images is seen in the right bottom corner in the picture. When trying on your dataset, the accuracy starts by about 60% and then it gets better by the time (reasonable). However when trying with my dataset it starts by 100% !! 
Attached also log file for the result. I'm using steps=5. I tried to cancel one (convolutional + pool) layer. I tried to change the filter size, the stride size. My pictures are in .png exension. I tried to change them to .jpg. I still have the same results. What would you suggest please?
![1](https://user-images.githubusercontent.com/40579781/43002008-97f7262a-8c27-11e8-8518-5207d5451fe6.png)
![2](https://user-images.githubusercontent.com/40579781/43002023-9fd4c5e6-8c27-11e8-9c38-841434b4303d.png)
[log.txt](https://github.com/kalaspuffar/tensorflow-data/files/2214045/log.txt)

**Comment by kalaspuffar** [comment]
Hi @rafaa1988 

Are you sure you've separated the training set and testing set? If you train a model on the same data that you test on then you will get 100% because you will overfit your model.

Another technique you can use to not overfit is adding a dropout layer and you can also change the learning rate and trainer.

Maybe you can get some ideas by watching this video:
https://www.youtube.com/watch?v=dONMcc0dXHw

or checking out some of the other videos in the Machine Learning playlist

https://www.youtube.com/playlist?list=PLP2v7zU48xOLt9Hqiu3j3PdBV5cKgqio7

Hope this helps

Best regards
Daniel

**Comment by rafaa1988** [comment]
Watched the videos. Thanks! It's overfitting as said. I checked My dataset and found that they were separated but repeated. So I deleted the repeated ones. I couldn't get more than 70% accuracy, so I increased the number of convolutional+max_pooling to 5, and used a variable learning_rate so it starts by 1e-3 and decreases for every new count. I got 95% after 4 hours. I'm using gpu-tensorflow.
I want to enhance the learning. So I think by decreasing the strides i can get better accuracy. I tried to reduce the stride from 2 to 1 but I got the following error:
ResourceExhaustedError (see above for traceback): OOM when allocating tensor with shape[3125824,128] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc
	 [[Node: layer_fc1/kernel/Initializer/random_uniform/RandomUniform = RandomUniform[T=DT_INT32, _class=["loc:@layer_fc1/kernel/Adam_1/Assign"], dtype=DT_FLOAT, seed=0, seed2=0, _device="/job:localhost/replica:0/task:0/device:GPU:0"](layer_fc1/kernel/Adam/Initializer/zeros/shape_as_tensor)]]

I tried to comment the first fully_connected layer, and got this error:
T:\src\github\tensorflow\tensorflow\stream_executor\cuda\cuda_dnn.cc:332] could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR

I tried to decrease the convolutional+max_pooling layers to 1. still have a problem. It seems that it doesn't accept a stride of 1.

**Comment by kalaspuffar** [comment]
Hi @rafaa1988 

Sorry, it took a while to answer. Still on vacation.

I think the problem here is memory usage in the GPU.
When it comes to GPUs you usually have a lot of compute power but a bit less memory.

So the problem you can have if you reduce the stride to early the amount of memory use might be to large to fit in the GPU.

So to explain the window and stride when doing pooling. You can have a 3x3 window you want to reduce your image and you move that window by 2 every time you read a value. If you move it only by 1 you get a lot more values in your output and many values might overlap. So a 2x2 window with a 1 stride might be better. But then again a 5x5 window with 3 strides might be good too. Depending on what you want to achieve.

Maybe just keep the stride for the first layer and reduce it further down.

You may also experiment with larger windows on the first layers.
I talk a bit about it in the video in the last reply.

Hope this post was helpful. Hard to help without something concrete to work with.

Best regards
Daniel

**Comment by rafaa1988** [comment]
Yes it's a problem with the memory. Thank you very much.
